libs/daemon/daemon.go

```diff
@@ -0,0 +1,117 @@
+package daemon
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"os/exec"
+	"strconv"
+)
+
+const DatabricksCliParentPid = "DATABRICKS_CLI_PARENT_PID"
+
+type Daemon struct {
+	// If provided, the child process will create a pid file at this path.
+	PidFilePath string
```

Comment:
Why do you need this?

Reply:
It's needed because of the way we do acceptance testing for the telemetry logger. In the telemetry acceptance tests we need to wait for the child process to exit before we can assert that all server API calls were indeed made.

https://github.com/databricks/cli/pull/2278